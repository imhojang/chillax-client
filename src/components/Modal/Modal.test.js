import React from 'react';
import { mount, shallow } from 'enzyme';
import Modal from './Modal';

describe('<Modal />', () => {
  let component = null;

  it('renders correctly', () => {
    component = shallow(<Modal />);
  });

  it('matches snapshot', () => {
    const handleClose = jest.fn();
    const wrapper = mount(<Modal show='true' handleClose={handleClose} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('props are passed down correctly', () => {
    const handleClose = jest.fn();
    const show = true;
    const wrapper = mount(<Modal show={show} handleClose={handleClose} />);
    expect(wrapper.props().show).toBe(show);
    expect(wrapper.props().handleClose).toBe(handleClose);
  });

  it('renders children prop correctly', () => {
    const children = <div className='children-div'></div>;
    const props = {
      children,
    };
    const wrapper = mount(<Modal {...props} />);
    const renderedChildren = wrapper.find('.children-div');
    expect(renderedChildren.hasClass('children-div')).toEqual(true);
  });
});
