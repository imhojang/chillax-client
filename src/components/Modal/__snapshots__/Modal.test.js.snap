// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> matches snapshot 1`] = `
<Modal
  handleClose={[MockFunction]}
  show="true"
>
  <div
    className="modal display-block"
  >
    <section
      className="modal-main"
    />
  </div>
</Modal>
`;
